src/sources/vector/mod.rs

```diff
@@ -50,6 +54,21 @@ impl proto::Service for Service {
             .map(Event::from)
             .collect();
 
+        if self.log_namespace == LogNamespace::Vector {
+            let now = Utc::now();
+            for event in &mut events {
+                if let Event::Log(ref mut log) = event {
+                    log.metadata_mut()
+                        .value_mut()
+                        .insert(path!("vector", "source_type"), VectorConfig::NAME);
+
+                    log.metadata_mut()
+                        .value_mut()
+                        .insert(path!("vector", "ingest_timestamp"), now);
```

Comment:
It was discussed offline that perhaps we might want to nest the metadata from the vector client so as to not override.

Wondering if anyone has opinions on that.

Reply:
I started to write out a whole thing here... but we intentionally don't send the metadata in sinks.

If a user wanted to curry the metadata they'd need to transform the events and move them into the event data - which is then just covered by the decoding.

Reply:
Ahh. So this isn't a valid concern then.